### PR TITLE
Remove branch encapsulation

### PR DESCRIPTION
**Pull request by EQuincerot**
Includes some documentation and a fix for https://github.com/lesfurets/JenkinsPipelineUnit/issues/21

### CODE REVIEW

File: src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy [hunk 1]
```diff
@@ -123,7 +123,15 @@ class PipelineTestHelper {
...
-            return intercepted.value.call(*args)
+            // When we use a library method, we should not spread the argument because we define a closure with a single
+            // argument. The arguments will be spread in this closure (See PipelineTestHelper#setGlobalVars)
+            // For other cases, we spread it before calling
+            // Note : InvokerHelper.invokeClosure(intercepted.value, args) is similar to intercepted.value.call(*args)
+            if (args?.size() > intercepted.value.maximumNumberOfParameters) {
+                return intercepted.value.call(args)
+            } else {
+                return intercepted.value.call(*args)
+            }
```
Comment (ozangunalp) [inline src/main/groovy/com/lesfurets/jenkins/unit/PipelineTestHelper.groovy]: We can extract a method for this

File: src/main/groovy/com/lesfurets/jenkins/unit/cps/PipelineTestHelperCPS.groovy [hunk 2]
```diff
@@ -54,7 +54,15 @@ class PipelineTestHelperCPS extends PipelineTestHelper {
...
-            return intercepted.value.call(*args)
+            // When we use a library method, we should not spread the argument because we define a closure with a single
+            // argument. The arguments will be spread in this closure (See PipelineTestHelper#setGlobalVars)
+            // For other cases, we spread it before calling
+            // Note : InvokerHelper.invokeClosure(intercepted.value, args) is similar to intercepted.value.call(*args)
+            if (args?.size() > intercepted.value.maximumNumberOfParameters) {
+                return intercepted.value.call(args)
+            } else {
+                return intercepted.value.call(*args)
+            }
```
Comment (ozangunalp) [inline src/main/groovy/com/lesfurets/jenkins/unit/cps/PipelineTestHelperCPS.groovy]: The extract method in `PipelineTestHelper` will avoid the duplication here

File: src/test/groovy/com/lesfurets/jenkins/TestSharedLibraryCPS.groovy [hunk 3]
```diff
@@ -73,7 +72,7 @@ class TestSharedLibraryCPS extends BasePipelineTestCPS {
             e.printStackTrace()
             exception = true
         }
-        assertThat(expected).isEqualTo(exception)
+        assertThat(exception).isEqualTo(expected)
```
Comment (ozangunalp) [inline src/test/groovy/com/lesfurets/jenkins/TestSharedLibraryCPS.groovy]: 👍

File: src/test/jenkins/job/library/libraryJob.jenkins [hunk 4]
```diff
@@ -8,14 +8,16 @@ acme.name = 'something'
 sh acme.name
 
 acme.caution('world')
-
+/*
 sayHello 'World'
 sayHello()
+*/
```
Comment (ozangunalp) [inline src/test/jenkins/job/library/libraryJob.jenkins]: Uncomment or remove commented block

File: src/test/resources/libs/commons@master/vars/sayHello.groovy [hunk 5]
```diff
@@ -1,5 +1,5 @@
 // vars/sayHello.groovy
-def call(String name = 'name', String otherName = null) {
+def call(String name, String otherName) {
```
Comment (ozangunalp) [inline src/test/resources/libs/commons@master/vars/sayHello.groovy]: Let's have default param values to tests include the test case.

File: src/main/groovy/com/lesfurets/jenkins/unit/cps/PipelineTestHelperCPS.groovy (hunk 2 above)
Comment (EQuincerot) [inline src/main/groovy/com/lesfurets/jenkins/unit/cps/PipelineTestHelperCPS.groovy]: Good idea!

File: src/test/jenkins/job/library/libraryJob.jenkins (hunk 4 above)
Comment (EQuincerot) [inline src/test/jenkins/job/library/libraryJob.jenkins]: Oops, my bad

File: src/test/resources/libs/commons@master/vars/sayHello.groovy (hunk 5 above)
Comment (EQuincerot) [inline src/test/resources/libs/commons@master/vars/sayHello.groovy]: Typo, it was supposed to be temporary.

**Pull request merged**